src/utils/ExtensionLoader.js

```diff
@@ -396,9 +397,17 @@ define(function (require, exports, module) {
         // Load *subset* of the usual builtin extensions list, and don't try to find any user/dev extensions
         if (brackets.inBrowser) {
             // Bramble: we have a more complex extension loading pattern based on URL params
-            var extensionList = BrambleExtensionLoader.getExtensionList(params);
+            var brambleExtensions = BrambleExtensionLoader.getExtensionList(params);
 
-            return Async.doInParallel(extensionList, function (item) {
+            return Async.doInParallel(brambleExtensions.extensionList, function (item) {
+                if (brambleExtensions.themes.indexOf(item) !== -1) {
+                    // XXXBramble: To make this more future proof,
+                    //  we need to read from package.json here to get
+                    //  a theme's metadata
```

Comment:
You can read the package.json using `require(['text!' + ..., function(packageJSON){...}` inside `BrambleExtensionLoader`.  Maybe make `BrambleExtensionLoader.getExtensionList(params)` be async, and you can get those files before you return the list.